### ISSUE TITLE
Error: ER_BAD_FIELD_ERROR: Unknown column 'undefined' in 'where clause'

### ISSUE THREAD

**Issue by Khanji** [mysqljs/mysql#1659]
## Full error message: 
```
Error: ER_BAD_FIELD_ERROR: Unknown column 'undefined' in 'where clause'
    at Query.Sequence._packetToError (..\mysql\lib\protocol\sequences\Sequence.js:52:14)
    at Query.ErrorPacket (..\mysql\lib\protocol\sequences\Query.js:77:18)
    at Protocol._parsePacket (..\mysql\lib\protocol\Protocol.js:280:23)
    at Parser.write ..\mysql\lib\protocol\Parser.js:75:12)
    at Protocol.write (..\mysql\lib\protocol\Protocol.js:39:16)
    at Socket.<anonymous> (..\mysql\lib\Connection.js:103:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:189:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    --------------------
    at Protocol._enqueue (..\mysql\lib\protocol\Protocol.js:141:48)
    at Connection.query ..\mysql\lib\Connection.js:208:25)
    at router.get (..\server\routes\api.js:77:14)
    at Layer.handle [as handle_request] (..\express\lib\router\layer.js:95:5)
    at next (..\express\lib\router\route.js:137:13)
    at Route.dispatch (..\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (..\express\lib\router\layer.js:95:5)
    at ..\express\lib\router\index.js:281:22
    at param (..\express\lib\router\index.js:354:14)
    at param (..\express\lib\router\index.js:365:14)
```
## Query:
```
SELECT p.*,
    IF ( o.name IS NULL, 'none', o.name ) AS owner_name,
    IF( u.name IS NULL, 'none', u.name ) AS creation_user_name
  FROM point p
    LEFT JOIN(SELECT u.id, u.name FROM user u) o ON p.owner_user_id = o.id
    INNER JOIN(SELECT u.id, u.name FROM user u) u ON p.creation_user_id = u.id
  WHERE p.creation_user_id = 1
  AND p.id = (SELECT MAX(id) FROM point);
```
confirmed that the query works on the server. Just not when I let msyql query it.

**Comment by dougwilson** [comment]
Hi @Khanji I assume that the query you're showing is what you're assuming that your code is building out, or is the query just a constant string? I ask because the error basically sounds like you're concatenating an undefined variable in the middle of your query. Is it possible to share the code you're using and the steps to reproduce with the code?

**Comment by Khanji** [comment]
Sure.

So this is the code I use to generate the query.

```
exports.createPoint = function (pId, fId, t, c, d, pValue, sValue, oUId = 0, cd, cUId, mUId) {
  return `
  INSERT INTO point(project_id,function_id,title,caption,description,priority_value
    ,status_value,owner_user_id,creation_date,creation_user_id,mutation_user_id)
    VALUES(${pId},${fId},'${t}','${c}','${d}',${pValue},${sValue},${oUId},'${cd}',${cUId},${
    mUId});
  SELECT p.*,
    IF ( o.name IS NULL, 'none', o.name ) AS owner_name,
    IF( u.name IS NULL, 'none', u.name ) AS creation_user_name
  FROM point p
  LEFT JOIN(SELECT u.id, u.name FROM user u) o ON p.owner_user_id = o.id
  INNER JOIN(SELECT u.id, u.name FROM user u) u ON p.creation_user_id = u.id
  WHERE p.creation_user_id = ${oUId}
  AND p.id = (SELECT MAX(id) FROM point); `;
};
```
This is the code where I execute the query and as you can see I console.logged the query before execution, to make sure if all the variables were passed trough. In addition, the dbLoginM adds multipleStatements: true to the connection parameters.
```
let connection = mysql.createConnection(dbLoginM);
  //start connection
  connection.connect();
  //query
  console.log(queries.createPoint(
    req.body.project_id,
    req.body.function_id,
    req.body.title,
    req.body.caption,
    req.body.description,
    req.body.priority_value,
    req.body.status_value,
    req.body.owner_user_id,
    req.body.creation_date,
    req.body.creation_user_id,
    req.body.mutation_user_id
  ));
  connection.query(queries.createPoint(
    req.body.project_id,
    req.body.function_id,
    req.body.title,
    req.body.caption,
    req.body.description,
    req.body.priority_value,
    req.body.status_value,
    req.body.owner_user_id,
    req.body.creation_date,
    req.body.creation_user_id,
    req.body.mutation_user_id
  ), function (error, results) {
    if (error) throw error;
    res.status(200).json(results[0]);
  });
  //end connection
  connection.end();
```
Just because I was wondering, I tested the select part of the query without the inserting part first. This works. But when I do it afterwards, I get the error.

**Comment by dougwilson** [comment]
Very weird, I'm not sure. I would love to dig into the issue, can you please provide all the following:

1. The version of Node.js you're using
2. The version of this module you're using
3. Complete code I can run to reproduce the issue (not just a snipplet)
4. Complete instructions on how to run your code to reproduce the issue.

**Comment by Khanji** [comment]
Will do. I will get back to you when I created an easy-to-make scenario to reproduce the error .

**Comment by dougwilson** [comment]
Another thing I noticed is that the query you posted above is only half the output of what your `createPoint` function generates. Is it possible to post what the entire output is where you get that error? I wonder if the error is in the `INSERT` part. I don't know the exact value for all those parameters to try it out myself.

**Comment by Khanji** [comment]
Here you go. By the way. The INSERT part does insert the values according to my expectations.
```
INSERT INTO point(project_id,function_id,title,caption,description,priority_value
    ,status_value,owner_user_id,creation_date,creation_user_id,mutation_user_id)
    VALUES(1,1,'Test this out','test','<p>&nbsp;test</p>
',3,6,1,'2017-03-11 20:29:09',1,1);
    SELECT p.*,IF(o.name IS NULL, 'none', o.name) AS owner_name, IF(u.name IS NULL, 'none', u.name) AS
    creation_user_name FROM point p
    LEFT JOIN(SELECT u.id,u.name FROM user u)o ON p.owner_user_id=o.id INNER JOIN(SELECT u.
    id,u.name FROM user u)u ON p.creation_user_id=u.id WHERE p.creation_user_id=1 AND p.id
    =(SELECT MAX(ID)FROM point);
```

**Comment by dougwilson** [comment]
Very strange. Another option, besides getting me code to reproduce, is that you can set `debug: true` in the connection options and then this module will write to STDERR all the data going to / from the MySQL server, so you should be able to see the exact query going across the wire when that error occurs.

Neither of your queries you provided gave me the error you provided above once I created a DDL that was good enough to work (I don't know what your actual DDL is).

**Comment by Khanji** [comment]
this is what the debug gives me:
I don't see anything wrong here. So this gets more mysterious..
TLDR:
`sql: '\n  INSERT INTO point(project_id,function_id,title,caption,description,priority_value\n    ,status_value,owner_user_id,creation_date,creation_user_id,mutation_user_id)\n    VALUES(1,1,\'yrdy\',\'sefse\',\'<p>fefeefeef</p>\n\',3,6,1,\'2017-03-11 21:34:57\',1,1);\n  SELECT p.*,\n    IF ( o.name IS NULL, \'none\', o.name ) AS owner_name,\n    IF( u.name IS NULL, \'none\', u.name ) AS creation_user_name\n  FROM point p\n  LEFT JOIN(SELECT u.id, u.name FROM user u) o ON p.owner_user_id = o.id\n  INNER JOIN(SELECT u.id, u.name FROM user u) u ON p.creation_user_id = u.id\n  WHERE p.creation_user_id = 1\n  AND p.id = (SELECT MAX(id) FROM point); ' }`
```
<-- HandshakeInitializationPacket
HandshakeInitializationPacket {
  protocolVersion: 10,
  serverVersion: '5.7.17-log',
  threadId: 2539,
  scrambleBuff1: <Buffer 30 59 0a 63 68 77 5b 48>,
  filler1: <Buffer 00>,
  serverCapabilities1: 63487,
  serverLanguage: 33,
  serverStatus: 2,
  serverCapabilities2: 33279,
  scrambleLength: 21,
  filler2: <Buffer 00 00 00 00 00 00 00 00 00 00>,
  scrambleBuff2: <Buffer 48 7c 09 6c 22 1c 17 3c 3a 48 78 08>,
  filler3: <Buffer 00>,
  pluginData: 'mysql_native_password',
  protocol41: true }

--> ClientAuthenticationPacket
ClientAuthenticationPacket {
  clientFlags: 521167,
  maxPacketSize: 0,
  charsetNumber: 33,
  filler: undefined,
  user: 'root',
  scrambleBuff: <Buffer 6b af 8f 66 1e ac 89 7d 3b d1 4b ce 8e 40 c2 7f d1 7e 72 53>,
  database: 'opoint',
  protocol41: true }

<-- OkPacket
OkPacket {
  fieldCount: 0,
  affectedRows: 0,
  insertId: 0,
  serverStatus: 2,
  warningCount: 0,
  message: '',
  protocol41: true,
  changedRows: 0 }

--> ComQueryPacket
ComQueryPacket {
  command: 3,
  sql: '\n  INSERT INTO point(project_id,function_id,title,caption,description,priority_value\n    ,status_value,owner_user_id,creation_date,creation_user_id,mutation_user_id)\n    VALUES(1,1,\'yrdy\',\'sefse\',\'<p>fefeefeef</p>\n\',3,6,1,\'2017-03-11 21:34:57\',1,1);\n  SELECT p.*,\n    IF ( o.name IS NULL, \'none\', o.name ) AS owner_name,\n    IF( u.name IS NULL, \'none\', u.name ) AS creation_user_name\n  FROM point p\n  LEFT JOIN(SELECT u.id, u.name FROM user u) o ON p.owner_user_id = o.id\n  INNER JOIN(SELECT u.id, u.name FROM user u) u ON p.creation_user_id = u.id\n  WHERE p.creation_user_id = 1\n  AND p.id = (SELECT MAX(id) FROM point); ' }

<-- OkPacket
OkPacket {
  fieldCount: 0,
  affectedRows: 1,
  insertId: 33,
  serverStatus: 10,
  warningCount: 0,
  message: '',
  protocol41: true,
  changedRows: 0 }

<-- ResultSetHeaderPacket
ResultSetHeaderPacket { fieldCount: 15, extra: undefined }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'id',
  orgName: 'id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 16899,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'project_id',
  orgName: 'project_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'function_id',
  orgName: 'function_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'title',
  orgName: 'title',
  charsetNr: 33,
  length: 765,
  type: 253,
  flags: 4097,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'caption',
  orgName: 'caption',
  charsetNr: 33,
  length: 765,
  type: 253,
  flags: 4097,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'description',
  orgName: 'description',
  charsetNr: 33,
  length: 196605,
  type: 252,
  flags: 4113,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'priority_value',
  orgName: 'priority_value',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'status_value',
  orgName: 'status_value',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'owner_user_id',
  orgName: 'owner_user_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 0,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'creation_date',
  orgName: 'creation_date',
  charsetNr: 63,
  length: 19,
  type: 7,
  flags: 9345,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'creation_user_id',
  orgName: 'creation_user_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'mutation_date',
  orgName: 'mutation_date',
  charsetNr: 63,
  length: 19,
  type: 7,
  flags: 1153,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'mutation_user_id',
  orgName: 'mutation_user_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: '',
  table: '',
  orgTable: '',
  name: 'owner_name',
  orgName: '',
  charsetNr: 33,
  length: 75,
  type: 253,
  flags: 0,
  decimals: 31,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: '',
  table: '',
  orgTable: '',
  name: 'creation_user_name',
  orgName: '',
  charsetNr: 33,
  length: 75,
  type: 253,
  flags: 1,
  decimals: 31,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- EofPacket
EofPacket {
  fieldCount: 254,
  warningCount: 0,
  serverStatus: 2,
  protocol41: true }

<-- RowDataPacket
RowDataPacket {
  id: 33,
  project_id: 1,
  function_id: 1,
  title: 'yrdy',
  caption: 'sefse',
  description: '<p>fefeefeef</p>\n',
  priority_value: 3,
  status_value: 6,
  owner_user_id: 1,
  creation_date: 2017-03-11T20:34:57.000Z,
  creation_user_id: 1,
  mutation_date: 2017-03-11T21:34:57.000Z,
  mutation_user_id: 1,
  owner_name: 'Ferdi',
  creation_user_name: 'Ferdi' }

<-- EofPacket
EofPacket {
  fieldCount: 254,
  warningCount: 0,
  serverStatus: 2,
  protocol41: true }

--> ComQuitPacket
ComQuitPacket { command: 1 }
```

**Comment by dougwilson** [comment]
Well, that debug output show a successful query, not a query that is resulting in the `ER_BAD_FIELD_ERROR: Unknown column 'undefined' in 'where clause'` error. I meant like you could use that to see what the query is when you get that error.

**Comment by Khanji** [comment]
It is like that! >.< Right after that output I get:
```
mysql\lib\protocol\Parser.js:79
        throw err; // Rethrow non-MySQL errors
        ^

Error: ER_BAD_FIELD_ERROR: Unknown column 'undefined' in 'where clause'
```

**Comment by Khanji** [comment]
Look: full output. To make sure you know, I know I am sending you the right information. 
By the way, I just want to let you know, I really appreciate your quick responses and support!!

```
API running on localhost:3000

  INSERT INTO point(project_id,function_id,title,caption,description,priority_value
    ,status_value,owner_user_id,creation_date,creation_user_id,mutation_user_id)
    VALUES(1,1,'yrdy','sefse','<p>fefeefeef</p>
',3,6,1,'2017-03-11 21:34:57',1,1);
  SELECT p.*,
    IF ( o.name IS NULL, 'none', o.name ) AS owner_name,
    IF( u.name IS NULL, 'none', u.name ) AS creation_user_name
  FROM point p
  LEFT JOIN(SELECT u.id, u.name FROM user u) o ON p.owner_user_id = o.id
  INNER JOIN(SELECT u.id, u.name FROM user u) u ON p.creation_user_id = u.id
  WHERE p.creation_user_id = 1
  AND p.id = (SELECT MAX(id) FROM point);
<-- HandshakeInitializationPacket
HandshakeInitializationPacket {
  protocolVersion: 10,
  serverVersion: '5.7.17-log',
  threadId: 2539,
  scrambleBuff1: <Buffer 30 59 0a 63 68 77 5b 48>,
  filler1: <Buffer 00>,
  serverCapabilities1: 63487,
  serverLanguage: 33,
  serverStatus: 2,
  serverCapabilities2: 33279,
  scrambleLength: 21,
  filler2: <Buffer 00 00 00 00 00 00 00 00 00 00>,
  scrambleBuff2: <Buffer 48 7c 09 6c 22 1c 17 3c 3a 48 78 08>,
  filler3: <Buffer 00>,
  pluginData: 'mysql_native_password',
  protocol41: true }

--> ClientAuthenticationPacket
ClientAuthenticationPacket {
  clientFlags: 521167,
  maxPacketSize: 0,
  charsetNumber: 33,
  filler: undefined,
  user: 'root',
  scrambleBuff: <Buffer 6b af 8f 66 1e ac 89 7d 3b d1 4b ce 8e 40 c2 7f d1 7e 72 53>,
  database: 'opoint',
  protocol41: true }

<-- OkPacket
OkPacket {
  fieldCount: 0,
  affectedRows: 0,
  insertId: 0,
  serverStatus: 2,
  warningCount: 0,
  message: '',
  protocol41: true,
  changedRows: 0 }

--> ComQueryPacket
ComQueryPacket {
  command: 3,
  sql: '\n  INSERT INTO point(project_id,function_id,title,caption,description,priority_value\n    ,status_value,owner_user_id,creation_date,creation_user_id,mutation_user_id)\n    VALUES(1,1,\'yrdy\',\'sefse\',\'<p>fefeefeef</p>\n\',3,6,1,\'2017-03-11 21:34:57\',1,1);\n  SELECT p.*,\n    IF ( o.name IS NULL, \'none\', o.name ) AS owner_name,\n    IF( u.name IS NULL, \'none\', u.name ) AS creation_user_name\n  FROM point p\n  LEFT JOIN(SELECT u.id, u.name FROM user u) o ON p.owner_user_id = o.id\n  INNER JOIN(SELECT u.id, u.name FROM user u) u ON p.creation_user_id = u.id\n  WHERE p.creation_user_id = 1\n  AND p.id = (SELECT MAX(id) FROM point); ' }

<-- OkPacket
OkPacket {
  fieldCount: 0,
  affectedRows: 1,
  insertId: 33,
  serverStatus: 10,
  warningCount: 0,
  message: '',
  protocol41: true,
  changedRows: 0 }

<-- ResultSetHeaderPacket
ResultSetHeaderPacket { fieldCount: 15, extra: undefined }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'id',
  orgName: 'id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 16899,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'project_id',
  orgName: 'project_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'function_id',
  orgName: 'function_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'title',
  orgName: 'title',
  charsetNr: 33,
  length: 765,
  type: 253,
  flags: 4097,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'caption',
  orgName: 'caption',
  charsetNr: 33,
  length: 765,
  type: 253,
  flags: 4097,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'description',
  orgName: 'description',
  charsetNr: 33,
  length: 196605,
  type: 252,
  flags: 4113,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'priority_value',
  orgName: 'priority_value',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'status_value',
  orgName: 'status_value',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'owner_user_id',
  orgName: 'owner_user_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 0,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'creation_date',
  orgName: 'creation_date',
  charsetNr: 63,
  length: 19,
  type: 7,
  flags: 9345,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'creation_user_id',
  orgName: 'creation_user_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'mutation_date',
  orgName: 'mutation_date',
  charsetNr: 63,
  length: 19,
  type: 7,
  flags: 1153,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: 'opoint',
  table: 'p',
  orgTable: 'point',
  name: 'mutation_user_id',
  orgName: 'mutation_user_id',
  charsetNr: 63,
  length: 11,
  type: 3,
  flags: 20489,
  decimals: 0,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: '',
  table: '',
  orgTable: '',
  name: 'owner_name',
  orgName: '',
  charsetNr: 33,
  length: 75,
  type: 253,
  flags: 0,
  decimals: 31,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- FieldPacket
FieldPacket {
  catalog: 'def',
  db: '',
  table: '',
  orgTable: '',
  name: 'creation_user_name',
  orgName: '',
  charsetNr: 33,
  length: 75,
  type: 253,
  flags: 1,
  decimals: 31,
  default: undefined,
  zeroFill: false,
  protocol41: true }

<-- EofPacket
EofPacket {
  fieldCount: 254,
  warningCount: 0,
  serverStatus: 2,
  protocol41: true }

<-- RowDataPacket
RowDataPacket {
  id: 33,
  project_id: 1,
  function_id: 1,
  title: 'yrdy',
  caption: 'sefse',
  description: '<p>fefeefeef</p>\n',
  priority_value: 3,
  status_value: 6,
  owner_user_id: 1,
  creation_date: 2017-03-11T20:34:57.000Z,
  creation_user_id: 1,
  mutation_date: 2017-03-11T21:34:57.000Z,
  mutation_user_id: 1,
  owner_name: 'Ferdi',
  creation_user_name: 'Ferdi' }

<-- EofPacket
EofPacket {
  fieldCount: 254,
  warningCount: 0,
  serverStatus: 2,
  protocol41: true }

--> ComQuitPacket
ComQuitPacket { command: 1 }

D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\Parser.js:79
        throw err; // Rethrow non-MySQL errors
        ^

Error: ER_BAD_FIELD_ERROR: Unknown column 'undefined' in 'where clause'
    at Query.Sequence._packetToError (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\sequences\Sequence.js:52:14)
    at Query.ErrorPacket (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\sequences\Query.js:77:18)
    at Protocol._parsePacket (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\Protocol.js:280:23)
    at Parser.write (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\Parser.js:75:12)
    at Protocol.write (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\Protocol.js:39:16)
    at Socket.<anonymous> (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\Connection.js:103:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:189:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    --------------------
    at Protocol._enqueue (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\protocol\Protocol.js:141:48)
    at Connection.query (D:\OneDrive\Raven Stone\oPoint\node_modules\mysql\lib\Connection.js:208:25)
    at router.get (D:\OneDrive\Raven Stone\oPoint\server\routes\api.js:78:14)
    at Layer.handle [as handle_request] (D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\layer.js:95:5)
    at next (D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\layer.js:95:5)
    at D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\index.js:281:22
    at param (D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\index.js:354:14)
    at param (D:\OneDrive\Raven Stone\oPoint\node_modules\express\lib\router\index.js:365:14)
```

**Comment by dougwilson** [comment]
That is weird indeed, because `Unknown column 'undefined' in 'where clause'` is a message directly from your MySQL server, and there is no error packet in the output above at all. Are you sure there is no other MySQL connection object in play here? I assume not, but just checking since I don't know what your code looks like exactly.

**Comment by Khanji** [comment]
> Are you sure there is no other MySQL connection object in play here? I assume not
Man, thank you for pointing this out. I checked the front-end which sends requests to the api. And it turns out that there was still some code there that should not be there anymore. It made get request to the api inside of the code that made a post request. 

Doug, I really want to thank you for your time and effort. I honestly appreciate it! I really thought it was a bug on your end, but as a usual thing in support-service, 80% of the time it is the user, not the software that makes mistakes...

**Comment by dougwilson** [comment]
Wow, awesome! I really didn't think that would have been it, but was almost getting to a loss of ideas so just decided to throw that possibility out there :)
